src/app/admin/volunteers/volunteers-details/volunteers-details.component.ts

```diff
@@ -51,4 +74,18 @@ export class VolunteersDetailsComponent implements OnInit, OnDestroy {
           .map((day) => this.translateService.instant(day))
           .join(', ');
   }
+
+  private loadDemands(id: string) {
+    console.log(id);
+    this.volunteerFacade.getVolunteerDemands(
+      { pageIndex: this.pageIndex, pageSize: this.pageSize },
+      id,
+    );
+  }
+
+  onPageChange($event: PageEvent) {
+    this.pageIndex = $event.pageIndex;
+    this.pageSize = $event.pageSize;
+    this.loadDemands(this.recordId);
```

Comment:
Can we replace this with below? And get rid totally of `recordId` and also remove `ngOnInit` at all, don't see any value from it.

```ts
this.volunteer$
  .pipe(takeUntil(this._destroy))
  .subscribe((v) => this.loadDemands(v._id));
```